app/frontend/src/features/profile/view/leaveReference/formSteps/submit/SubmitReference.tsx

```diff
@@ -0,0 +1,185 @@
+
+export interface SubmitReferenceProps {
+  user: User.AsObject;
+  referenceData: ReferenceContextFormData;
+}
+
+export default function SubmitReference({
+  user,
+  referenceData,
+}: SubmitReferenceProps) {
+  const {
+    writeFriendReference,
+    reset: resetFriendReferenceWriting,
+    error: friendReferenceError,
+    isLoading: isFriendReferenceLoading,
+  } = useWriteFriendReference(user.userId);
+  const {
+    writeHostRequestReference,
+    reset: resetHostRequestReferenceWriting,
+    error: hostRequestReferenceError,
+    isLoading: isHostRequestReferenceLoading,
+  } = useWriteHostReference(user.userId);
+
+  const { referenceType, hostRequest } = useParams<{
+    referenceType: string;
+    hostRequest?: string;
+  }>();
+  const theme = useTheme();
+  const classes = useReferenceStyles();
+  const history = useHistory();
+  const [isOpen, setIsOpen] = useState(false);
+  const isSmOrWider = useMediaQuery(theme.breakpoints.up("sm"));
+  const { handleSubmit } = useForm<ReferenceContextFormData>();
+
+  const onFriendReferenceSubmit = () => {
+    const formData: WriteFriendReferenceInput =
+      referenceData.wasAppropriate === "true"
+        ? {
+            toUserId: user.userId,
+            wasAppropriate: true,
+            text: referenceData.text,
+            rating: referenceData.rating,
+          }
+        : {
+            toUserId: user.userId,
+            wasAppropriate: false,
+            text: referenceData.text,
+            rating: referenceData.rating,
+          };
+
+    resetFriendReferenceWriting();
+    writeFriendReference(
+      {
+        referenceData: formData,
+      },
+      {
+        onSuccess: () => {
+          history.push(`${userRoute}/${user.username}/references`);
+        },
+      }
+    );
+    window.scroll({ top: 0 });
+  };
+
+  const onHostReferenceSubmit = () => {
+    if (hostRequest && +hostRequest !== undefined) {
+      const formData: WriteHostRequestReferenceInput =
+        referenceData.wasAppropriate === "true"
+          ? {
+              hostRequestId: +hostRequest,
+              wasAppropriate: true,
+              text: referenceData.text,
+              rating: referenceData.rating,
+            }
+          : {
+              hostRequestId: +hostRequest,
+              wasAppropriate: false,
+              text: referenceData.text,
+              rating: referenceData.rating,
+            };
+
+      resetHostRequestReferenceWriting();
+      writeHostRequestReference(
+        {
+          referenceData: formData,
+        },
+        {
+          onSuccess: () => {
+            setIsOpen(true);
+          },
+        }
+      );
+      window.scroll({ top: 0 });
+    }
+  };
+
+  const redirectToHome = () => {
+    history.push(`${baseRoute}`);
+  };
+
+  let onSubmit: () => void;
+  referenceType === referenceTypeRoute[ReferenceType.REFERENCE_TYPE_FRIEND]
+    ? (onSubmit = onFriendReferenceSubmit)
+    : (onSubmit = onHostReferenceSubmit);
```

Comment:
```suggestion
  const onSubmit =
  referenceType === referenceTypeRoute[ReferenceType.REFERENCE_TYPE_FRIEND]
    ? onFriendReferenceSubmit
    : onHostReferenceSubmit;
```
Again, might want to Prettier format if you apply directly. This would eliminate the need to use `let` and explicitly type the variable